### PR TITLE
Fix invalid allocate_like() and empty_like() tests.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Fixes:  https://github.com/rapidsai/cudf/issues/11247

These tests were using `cudf::test::expect_column_properties_equal()` which explicitly compares null counts.  However the tests were using uninitialized memory for the null mask, so it experienced random failures depending on what was returned from the memory manager.   The fix is to simply use a custom set of checks specific to these tests rather than the internal cudf functionality.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Comment by bdice** [comment]
I think we should evaluate/discuss an alternative solution here. I would argue that the test construction is flawed, not the assertions about properties. I would prefer to use `mask_policy::UNALLOCATED` on the input, which would be retained (as unallocated) by the default mask allocation policy of `cudf::allocate_like`. Then the properties (including null count) would match up.

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by nvdbaranec** [comment]
At @bdice's suggestion, I changed the fix so that we just use ALL_VALID for the comparison instead of leaving it uninitialized.

File: cpp/tests/copying/utility_tests.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include <cudf/copying.hpp>
 #include <cudf/strings/detail/utilities.hpp>
 #include <cudf/table/table.hpp>
+#include <cudf/utilities/type_checks.hpp>
```
Comment (bdice) [inline cpp/tests/copying/utility_tests.cpp:21]: The only function declared in this header is `column_types_equal`, which does not appear to be used in this file. Can we remove this include? https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/include/cudf/utilities/type_checks.hpp

**Review by bdice — APPROVED** [review]
One small request -- otherwise this looks great! The test design has clearer intent with this change. Thanks for the consideration @nvdbaranec!

**Review by hyperbolic2346 — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

**Comment by nvdbaranec** [comment]
@gpucibot merge

**Pull request merged**